Tests/OptimizelyTests-DataModel/SemanticVersionTests.swift

```diff
@@ -0,0 +1,121 @@
+        let target = "2.0" as SemanticVersion
+        let version = "2.0.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) == 0)
+    }
+
+    func testTargetFullStringTargetLess() {
+        let target = "2.0.0" as SemanticVersion
+        let version = "2.0.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) > 0)
+    }
+
+    func testTargetFullStringTargetMore() {
+        let target = "2.0.1" as SemanticVersion
+        let version = "2.0.0" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) < 0)
+    }
+
+    func testTargetFullStringTargetEq() {
+        let target = "2.0.0" as SemanticVersion
+        let version = "2.0.0" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) == 0)
+    }
+    func testTargetMajorPartGreater() {
+        let target = "3.0" as SemanticVersion
+        let version = "2.0.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) < 0)
+    }
+
+    func testTargetMajorPartLess() {
+        let target = "2.0" as SemanticVersion
+        let version = "3.0.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) > 0)
+    }
+
+    func testTargetMinorPartGreater() {
+        let target = "2.3" as SemanticVersion
+        let version = "2.0.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) < 0)
+    }
+
+    func testTargetMinorPartLess() {
+        let target = "2.0" as SemanticVersion
+        let version = "2.9.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) > 0)
+    }
+
+    func testTargetMinorPartEqual() {
+        let target = "2.9" as SemanticVersion
+        let version = "2.9.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) == 0)
+    }
+
+    func testTargetPatchGreater() {
+        let target = "2.3.5" as SemanticVersion
+        let version = "2.3.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) < 0)
+    }
+
+    func testTargetPatchLess() {
+        let target = "2.9.0" as SemanticVersion
+        let version = "2.9.1" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) > 0)
+    }
+
+    func testTargetPatchEqual() {
+        let target = "2.9.9" as SemanticVersion
+        let version = "2.9.9" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) == 0)
+    }
+
+    func testTargetPatchWithBetaTagEqual() {
+        let target = "2.9.9-beta" as SemanticVersion
+        let version = "2.9.9-beta" as SemanticVersion
+        
+        XCTAssert(version.compareVersion(targetedVersion: target) == 0)
+    }
+
+    func testPartialVersionEqual() {
```

Comment:
Can we add tests:
- 1.0.0 = 1.0.0-beta
- 1.0.0 < 1.1.0-beta
- 1.2.0-beta > 1.1.0
- 1.2.0-beta < 1.2.1

Reply:
Also need to test with different "beta" rules: 1.2.0.beta, 1.2.0-beta

Reply:
1.2.0.beta is not valid semver